### ISSUE TITLE
Poor training results of PSPNet

### ISSUE THREAD

**Issue by endaoguansanlu** [open-mmlab/mmsegmentation#2436]
Hello, thank you very much for your outstanding work! I tried to retrain PSPNet on Cityscapes, but I only got an unsatisfactory result. At present, I don't know what caused this result. Can you give me some guidance? Looking forward to your reply!
My mmsegmentation version is 0.13.0, and my mmcv version is 1.4.2

+---------------+-------+-------+
|     Class     |  IoU  |  Acc  |
+---------------+-------+-------+
|      road     | 97.96 | 99.33 |
|    sidewalk   | 84.53 | 89.34 |
|    building   | 91.86 | 96.95 |
|      wall     | 35.61 | 38.15 |
|     fence     | 50.58 | 54.63 |
|      pole     | 68.44 |  79.7 |
| traffic light | 73.22 | 80.69 |
|  traffic sign | 80.43 | 86.35 |
|   vegetation  | 92.36 | 97.46 |
|    terrain    | 62.95 | 67.36 |
|      sky      | 94.44 | 97.43 |
|     person    | 84.16 |  93.3 |
|     rider     | 64.22 | 74.38 |
|      car      | 93.34 | 98.29 |
|     truck     | 69.07 | 71.39 |
|      bus      | 56.38 | 56.77 |
|     train     | 33.46 | 33.98 |
|   motorcycle  | 51.93 | 55.93 |
|    bicycle    | 79.83 | 90.61 |
+---------------+-------+-------+
2022-12-25 05:11:30,979 - mmseg - INFO - Summary:
2022-12-25 05:11:30,979 - mmseg - INFO - 
+-------+-------+-------+
|  aAcc |  mIoU |  mAcc |
+-------+-------+-------+
| 95.76 | 71.83 | 76.95 |
+-------+-------+-------+
Here is my config:

`_base_ = [
    '../_base_/models/pspnet_r50-d8.py', '../_base_/datasets/cityscapes.py',
    '../_base_/default_runtime.py', '../_base_/schedules/schedule_160k.py'
]

model = dict(pretrained='open-mmlab://resnet101_v1c', backbone=dict(depth=101))

# dataset settings
dataset_type = 'CityscapesDataset'
data_root = '/home/liz/data/Cityscapes/'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
crop_size = (512, 1024)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations'),
    dict(type='Resize', img_scale=(2048, 1024), ratio_range=(0.5, 2.0)),
    dict(type='RandomCrop', crop_size=crop_size, cat_max_ratio=0.75),
    dict(type='RandomFlip', prob=0.5),
    dict(type='PhotoMetricDistortion'),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='Pad', size=crop_size, pad_val=0, seg_pad_val=255),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_semantic_seg']),
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(2048, 1024),
        img_ratios=[0.5, 0.75, 1.0, 1.25, 1.5, 1.75],
        flip=True,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(type='Normalize', **img_norm_cfg),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img']),
        ])
]
data = dict(
    samples_per_gpu=2,
    workers_per_gpu=2,
    train=dict(
            type=dataset_type,
            data_root=data_root,
            img_dir='leftImg8bit/train',
            ann_dir='gtFine/train',
            pipeline=train_pipeline),
    val=dict(
        type=dataset_type,
        data_root=data_root,
        img_dir='leftImg8bit/val',
        ann_dir='gtFine/val',
        pipeline=test_pipeline),
    test=dict(
        type=dataset_type,
        data_root=data_root,
        img_dir='leftImg8bit/val',
        ann_dir='gtFine/val',
        pipeline=test_pipeline))`

**Comment by MengzhangLI** [comment]
Hi, I think it is caused by GPU numebers, in Cityscapes dataset, default setting is 4 GPUs x 2 samplers_per_gpu = 8 total batch sizes.

**Comment by endaoguansanlu** [comment]
> Hi, I think it is caused by GPU numebers, in Cityscapes dataset, default setting is 4 GPUs x 2 samplers_per_gpu = 8 total batch sizes.

Thank you for your answer! I also have a question. If I train 8 batchsizes on a single GPU, will the effect be improved?

**Comment by MengzhangLI** [comment]
> > Hi, I think it is caused by GPU numebers, in Cityscapes dataset, default setting is 4 GPUs x 2 samplers_per_gpu = 8 total batch sizes.
> 
> Thank you for your answer! I also have a question. If I train 8 batchsizes on a single GPU, will the effect be improved?

Technically, no difference in that total batch sizes are the same. But 1 GPU training would be much slower than 4 GPUs.
